spring-cloud-gcp-starters/spring-cloud-gcp-starter-config/src/main/java/org/springframework/cloud/gcp/config/GoogleConfigPropertySourceLocator.java

```diff
@@ -76,7 +76,7 @@ public GoogleConfigPropertySourceLocator(GcpProjectIdProvider projectIdProvider,
 		if (gcpConfigProperties.isEnabled()) {
 			Assert.notNull(credentialsProvider, "Credentials provider cannot be null");
 			Assert.notNull(projectIdProvider, "Project ID provider cannot be null");
-			this.credentials = gcpConfigProperties.getCredentials() != null
+			this.credentials = gcpConfigProperties.getCredentials().getLocation() != null
```

Comment:
Sounds like we should also add an assertion to check for a null `gcpConfigProperties.getCredentials()`?

Reply:
I guess if credentials are `final` and we're setting them right away, they can't be null.